SocialCareCaseViewerApi.Tests/V1/Gateways/DatabaseGateway.Tests.cs

```diff
@@ -272,6 +271,51 @@ public void UpdateWorkerUpdatesAnExistingWorkerInDatabase()
             UpdatedWorkerShouldBeEquivalentToRequest(updatedWorker, request, worker, team);
         }
 
+        [Test]
+        public void UpdateWorkerUpdatesTheTeamSetOnAnyAllocations()
+        {
+            var worker = TestHelpers.CreateWorker(hasAllocations: false, hasWorkerTeams: false, id: 123);
+            var workerTeam = TestHelpers.CreateWorkerTeam(worker.Id);
+            worker.WorkerTeams = new List<WorkerTeam> { workerTeam };
+
+            DatabaseContext.Workers.Add(worker);
+            DatabaseContext.WorkerTeams.Add(workerTeam);
+
+            var differentTeam = TestHelpers.CreateTeam(teamId: 2000, name: "different team X", context: worker.ContextFlag);
+            DatabaseContext.Teams.Add(differentTeam);
+
+            DatabaseContext.SaveChanges();
+
+            (var createAllocationRequest, _, var createdByWorker, var person, _) = TestHelpers.CreateAllocationRequest(workerId: worker.Id, teamId: workerTeam.TeamId);
+            DatabaseContext.Workers.Add(createdByWorker);
+            DatabaseContext.Persons.Add(person);
+            DatabaseContext.SaveChanges();
+
+            _classUnderTest.CreateAllocation(createAllocationRequest);
+
+            var originalWorker = _classUnderTest.GetWorkerByWorkerId(worker.Id);
+            var originalTeam = DatabaseContext.WorkerTeams.Where(x => x.WorkerId == originalWorker.Id).Single().Team;
+
+            originalWorker.WorkerTeams?.Count.Should().Be(1);
+            originalWorker.Allocations.Count.Should().Be(1);
+
+            var originalAllocation = _classUnderTest.SelectAllocations(0, workerId: originalWorker.Id);
+
+            originalAllocation.Count.Should().Be(1);
+            originalAllocation.Single().AllocatedWorkerTeam.Should().Be(originalTeam.Name);
+
+            var updateRequest = TestHelpers.CreateUpdateWorkersRequest(teamId: differentTeam.Id, teamName: differentTeam.Name,
+                workerId: worker.Id, firstName: worker.FirstName, lastName: worker.LastName, role: worker.Role, contextFlag: worker.ContextFlag);
+
+            _classUnderTest.UpdateWorker(updateRequest);
+
+            var getUpdatedWorker = _classUnderTest.GetWorkerByWorkerId(worker.Id);
```

Comment:
Should we have a test against `getUpdatedWorker` and also `originalWorker`, show that `originalWorker.WorkerTeams` is a length of 1 and that the teamId matches the `workerTeam`. Then test that the `updatedWorker` still has `WorkerTeams` of length 1 but the teamId now matches `differentTeam`.

Reply:
We could 🤔 
We did this previously but the test was returning a false positive. I think I remember reading that entity framework would remember the changes on an entity if it had touched it once or something like that. This false positive is what inspired having an integration test but does make sense that we can also assert at the gateway level, that what we want happens here too.

Reply:
Done in fe36ffb